go/vt/sqlparser/ast.go

```diff
@@ -1479,10 +1479,10 @@ type Show struct {
 
 // Format formats the node.
 func (node *Show) Format(buf *TrackedBuffer) {
-	if (node.Type == "tables" || node.Type == "columns") && node.ShowTablesOpt != nil {
+	if (node.Type == "tables" || node.Type == "columns" || node.Type == "fields") && node.ShowTablesOpt != nil {
 		opt := node.ShowTablesOpt
 		buf.Myprintf("show %s%s", opt.Full, node.Type)
-		if node.Type == "columns" && node.HasOnTable() {
+		if node.Type == "columns" || node.Type == "fields" && node.HasOnTable() {
```

Comment:
This needs to be parenthesized. However, the better option (looking at the grammar) is:
```
if node.HasOnTable() {
  if node.Type == "vindexes" {
    buf.Myprintf(" on %v", node.OnTable)
  } else {
    buf.Myprintf(" from %v", node.OnTable)
  }
}
```

and delete the `HasOnTable` condition below.

Reply:
I am not quite sure I follow. To enter this block `node.Type` needs to be tables, columes, or fields so it would never get to the vindexes branch. Also at the end it requires opt to be non nil which I gather from reading the code now is not necessary